app/src/main/java/com/duckduckgo/app/downloads/DownloadsActivity.kt

```diff
@@ -158,18 +161,30 @@ class DownloadsActivity : DuckDuckGoActivity() {
                     BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_ACTION -> viewModel.insert(command.items)
                     BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_SWIPE,
                     BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_MANUAL,
-                    BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_TIMEOUT -> viewModel.deleteFilesFromDisk(command.items)
+                    BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_TIMEOUT -> handleDeleteAll(command.items)
                     BaseTransientBottomBar.BaseCallback.DISMISS_EVENT_CONSECUTIVE -> { /* noop */ }
                 }
             }
         }).show()
     }
 
+    private fun handleDeleteAll(items: List<DownloadItem>) {
+        viewModel.deleteFilesFromDisk(items)
+        // Remove all unfinished downloads from DownloadManager.
+        (getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager?)?.let { manager ->
```

Comment:
Should this logic be in the `Activity`? Maybe the `ViewModel` can have `DownloadManager` as a collaborator?

Reply:
Updated, thanks!